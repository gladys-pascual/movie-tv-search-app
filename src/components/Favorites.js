import React, { useState, useEffect } from "react";
import ShowCard from "./ShowCard";
import { Link } from "react-router-dom";
import NoFavorites from "./NoFavorites";
import Loading from "./Loading";

const Favorites = (userDetails) => {
  const [favoriteMovies, setFavoriteMovies] = useState({
    results: [],
  });
  const [favoriteTvs, setFavoriteTvs] = useState({
    results: [],
  });

  const [loading, setLoading] = useState(true);

  const sessionId = localStorage.getItem("session_id");

  useEffect(() => {
    fetch(
      `https://api.themoviedb.org/3/account/${userDetails.id}/favorite/movies?api_key=${process.env.REACT_APP_API_KEY}&session_id=${sessionId}&language=en-US&sort_by=created_at.asc&page=1`
    )
      .then((response) => response.json())
      .then((response) => {
        // setFavoriteMovies({ results: [] });
        setFavoriteMovies(response);
        setLoading(false);
      })
      .catch((error) => console.log("Error fetching and parsing data", error));
  }, [userDetails.id, sessionId]);

  const favoriteMovie = favoriteMovies.results.map((movie) => {
    return (
      <Link to={`/movie/${movie.id}`} key={`tv-${movie.id}`}>
        <ShowCard {...movie} key={movie.id} media_type="movie" />
      </Link>
    );
  });

  useEffect(() => {
    fetch(
      `https://api.themoviedb.org/3/account/${userDetails.id}/favorite/tv?api_key=${process.env.REACT_APP_API_KEY}&language=en-US&session_id=${sessionId}&sort_by=created_at.asc&page=1`
    )
      .then((response) => response.json())
      .then((response) => {
        // setFavoriteTvs({ results: [] });
        setFavoriteTvs(response);
        setLoading(false);
      })
      .catch((error) => console.log("Error fetching and parsing data", error));
  }, [userDetails.id, sessionId]);

  console.log(favoriteTvs);

  const favoriteTv = favoriteTvs.results.map((tv) => {
    return (
      <Link to={`/tv/${tv.id}`} key={`tv-${tv.id}`}>
        <ShowCard {...tv} key={tv.id} media_type="tv" />
      </Link>
    );
  });

  if (loading) {
    return <Loading />;
  }

  return !favoriteMovie.length && !favoriteTv.length ? (
    <NoFavorites />
  ) : (
    <section className="favorites">
      <h2 className="favorites-heading"> Here are your top picks: </h2>
      {favoriteMovie.length && (
        <div className="favorite-movie results">
          <h3 className="favorite-movie-heading"> Your favorite movies </h3>
          <div className="show-list">{favoriteMovie}</div>
        </div>
      )}
      {favoriteTv.length && (
        <div className="favorite-tv results">
          <h3 className="favorite-tv-heading"> Your favorite TV shows </h3>
          <div className="show-list">{favoriteTv}</div>
        </div>
      )}
    </section>
  );
};

export default Favorites;
