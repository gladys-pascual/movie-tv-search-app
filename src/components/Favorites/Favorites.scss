@import "../../assets/variables.scss";

.favorites {
  .favorites-heading {
    color: $text-color;
    text-align: center;
    font-size: 24px;
  }
  .results {
    .favorite-movie-heading,
    .favorite-tv-heading {
      color: $text-color;
      padding-left: 10vw;
      font-size: 20px;
      font-style: italic;
    }
  }
}
