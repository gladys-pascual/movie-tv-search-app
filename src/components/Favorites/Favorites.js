import ShowCard from "../ShowCard/ShowCard";
import { Link } from "react-router-dom";
import NoFavorites from "../NoFavorites/NoFavorites";
import Loading from "../Loading/Loading.js";
import React from "react";
import "./Favorites.scss";

const Favorites = ({ favoriteMovies, favoriteTvs, loading }) => {
  const favoriteMovie = favoriteMovies.results.map((movie) => {
    return (
      <Link to={`/movie/${movie.id}`} key={`tv-${movie.id}`}>
        <ShowCard {...movie} key={movie.id} media_type="movie" />
      </Link>
    );
  });

  const favoriteTv = favoriteTvs.results.map((tv) => {
    return (
      <Link to={`/tv/${tv.id}`} key={`tv-${tv.id}`}>
        <ShowCard {...tv} key={tv.id} media_type="tv" />
      </Link>
    );
  });

  if (loading) {
    return <Loading />;
  }

  return !favoriteMovie.length && !favoriteTv.length ? (
    <NoFavorites />
  ) : (
    <section className="favorites">
      <h2 className="favorites-heading"> Here are your top picks: </h2>
      {favoriteMovie.length && (
        <div className="favorite-movie results">
          <h3 className="favorite-movie-heading"> Your favorite movies </h3>
          <div className="show-list">{favoriteMovie}</div>
        </div>
      )}
      {favoriteTv.length && (
        <div className="favorite-tv results">
          <h3 className="favorite-tv-heading"> Your favorite TV shows </h3>
          <div className="show-list">{favoriteTv}</div>
        </div>
      )}
    </section>
  );
};

export default Favorites;
