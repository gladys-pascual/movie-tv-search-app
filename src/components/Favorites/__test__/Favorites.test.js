import React from "react";
import { render } from "@testing-library/react";
import Favorites from "../Favorites";
import { BrowserRouter } from "react-router-dom";

test("render Favorites component as expected", () => {
  const props = {
    favoriteMovies: {
      results: [
        {
          original_language: "en",
          original_title: "Harry Potter and the Philosopher's Stone",
          poster_path: "/wuMc08IPKEatf9rnMNXvIDxqP4W.jpg",
          video: false,
          vote_average: 7.9,
          overview:
            "Harry Potter has lived under the stairs at his aunt and uncle's house his whole life. But on his 11th birthday, he learns he's a powerful wizard -- with a place waiting for him at the Hogwarts School of Witchcraft and Wizardry. As he learns to harness his newfound powers with the help of the school's kindly headmaster, Harry uncovers the truth about his parents' deaths -- and about the villain who's to blame.",
          release_date: "2001-11-16",
          vote_count: 19508,
          title: "Harry Potter and the Philosopher's Stone",
          adult: false,
          backdrop_path: "/hziiv14OpD73u9gAak4XDDfBKa2.jpg",
          id: 671,
          genre_ids: [12, 14],
          popularity: 173.241,
        },
      ],
    },
    favoriteTvs: {
      results: [
        {
          overview:
            "A 10-part documentary chronicling the untold story of Michael Jordan and the Chicago Bulls dynasty with rare, never-before-seen footage and sound from the 1997-98 championship season – plus over 100 interviews with famous figures and basketball’s biggest names.",
          vote_count: 735,
          backdrop_path: "/kY0h95L73t7a6ev6Rv0aHSCtN7y.jpg",
          id: 79525,
          genre_ids: [99],
          origin_country: ["US"],
          original_language: "en",
          vote_average: 8.2,
          poster_path: "/oVf4xGGbDtwVHiKn8uTuSriY7PH.jpg",
          first_air_date: "2020-04-19",
          original_name: "The Last Dance",
          name: "The Last Dance",
          popularity: 24.461,
        },
      ],
    },
    loading: false,
  };
  const { container } = render(
    <BrowserRouter>
      <Favorites {...props} />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
