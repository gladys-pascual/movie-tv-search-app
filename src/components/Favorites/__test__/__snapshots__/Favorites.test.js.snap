// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Favorites component as expected 1`] = `
<div>
  <section
    class="favorites"
  >
    <h2
      class="favorites-heading"
    >
       Here are your top picks: 
    </h2>
    <div
      class="favorite-movie results"
    >
      <h3
        class="favorite-movie-heading"
      >
         Your favorite movies 
      </h3>
      <div
        class="show-list"
      >
        <a
          href="/movie/671"
        >
          <div
            class="show-card"
          >
            <div
              class="top-wrapper"
            >
              <div
                class="img-wrapper"
              >
                <img
                  alt="Harry Potter and the Philosopher's Stone"
                  src="https://image.tmdb.org/t/p/w220_and_h330_face/wuMc08IPKEatf9rnMNXvIDxqP4W.jpg"
                />
              </div>
              <div
                class="text-wrapper"
              >
                <h3>
                  Harry Potter and the Philosopher's Stone
                </h3>
                <p
                  class="date"
                >
                   
                  Nov 16 2001
                   
                </p>
              </div>
            </div>
            <div
              class="rating-wrapper"
            >
              <div
                class="icon"
              >
                <img
                  alt="movie"
                  src="https://img.icons8.com/ios-filled/20/000000/movie.png"
                />
              </div>
              <div
                class="react-rater"
              >
                <div>
                  <div
                    class="react-rater-star is-disabled is-active"
                  >
                    ★
                  </div>
                </div>
                <div>
                  <div
                    class="react-rater-star is-disabled is-active"
                  >
                    ★
                  </div>
                </div>
                <div>
                  <div
                    class="react-rater-star is-disabled is-active"
                  >
                    ★
                  </div>
                </div>
                <div>
                  <div
                    class="react-rater-star is-disabled is-active-half"
                  >
                    ★
                  </div>
                </div>
                <div>
                  <div
                    class="react-rater-star is-disabled"
                  >
                    ★
                  </div>
                </div>
              </div>
              <p
                class="rating"
              >
                7.9
              </p>
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      class="favorite-tv results"
    >
      <h3
        class="favorite-tv-heading"
      >
         Your favorite TV shows 
      </h3>
      <div
        class="show-list"
      >
        <a
          href="/tv/79525"
        >
          <div
            class="show-card"
          >
            <div
              class="top-wrapper"
            >
              <div
                class="img-wrapper"
              >
                <img
                  alt="The Last Dance"
                  src="https://image.tmdb.org/t/p/w220_and_h330_face/oVf4xGGbDtwVHiKn8uTuSriY7PH.jpg"
                />
              </div>
              <div
                class="text-wrapper"
              >
                <h3>
                  The Last Dance
                </h3>
                <p
                  class="date"
                >
                   
                  Apr 19 2020
                   
                </p>
              </div>
            </div>
            <div
              class="rating-wrapper"
            >
              <div
                class="icon"
              >
                <img
                  alt="tv"
                  src="https://img.icons8.com/android/20/000000/retro-tv.png"
                />
              </div>
              <div
                class="react-rater"
              >
                <div>
                  <div
                    class="react-rater-star is-disabled is-active"
                  >
                    ★
                  </div>
                </div>
                <div>
                  <div
                    class="react-rater-star is-disabled is-active"
                  >
                    ★
                  </div>
                </div>
                <div>
                  <div
                    class="react-rater-star is-disabled is-active"
                  >
                    ★
                  </div>
                </div>
                <div>
                  <div
                    class="react-rater-star is-disabled is-active"
                  >
                    ★
                  </div>
                </div>
                <div>
                  <div
                    class="react-rater-star is-disabled"
                  >
                    ★
                  </div>
                </div>
              </div>
              <p
                class="rating"
              >
                8.2
              </p>
            </div>
          </div>
        </a>
      </div>
    </div>
  </section>
</div>
`;
